### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by tarasulo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,40 @@
-# onlineshop;
-welcome
+# online-shop
```
Comment (Sofasmile) [inline README.md]:
```suggestion
# Online-shop
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,40 @@
...
+![Language grade: Java](images/onlinestore.jpg)
```
Comment (Sofasmile) [inline README.md:2]: check if this image is showing correctly

File: settings.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
```
Comment (Sofasmile) [inline settings.xml]: do we need this file here? remove it from repository

File: src/main/java/internet/shop/controller/AddToBucketController.java [hunk 4]
```diff
@@ -2,27 +2,37 @@
...
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+
 import org.apache.log4j.Logger;
 
 public class AddToBucketController extends HttpServlet {
     private static Logger logger = Logger.getLogger(AddToBucketController.class);
     @Inject
     static BucketService bucketService;
```
Comment (Sofasmile) [inline src/main/java/internet/shop/controller/AddToBucketController.java]: add access modifier

File: src/main/java/internet/shop/dao/hibernate/BucketDaoHibernateImpl.java [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class BucketDaoHibernateImpl implements BucketDao {
+    private static Logger logger = Logger.getLogger(BucketDaoHibernateImpl.class);
+    @Inject
+    private static ItemService itemService;
```
Comment (Sofasmile) [inline src/main/java/internet/shop/dao/hibernate/BucketDaoHibernateImpl.java]: can we use service in dao layer?

File: src/main/java/internet/shop/dao/hibernate/UserDaoHibernateImpl.java [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+                session.close();
+            }
+        }
+        user.setId(userId);
+        return user;
+    }
+
+    @Override
+    public User get(Long id) {
+
```
Comment (Sofasmile) [inline src/main/java/internet/shop/dao/hibernate/UserDaoHibernateImpl.java:47]: remove empty line

File: src/main/java/internet/shop/dao/hibernate/UserDaoHibernateImpl.java [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    @Override
+    public User login(String login, String psw)
+            throws AuthenticationException {
+        User user = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query query = session.createQuery("FROM User WHERE login=:login");
+            query.setParameter("login", login);
+            user = (User) query.uniqueResult();
+            if (user.getPassword().equals(HashUtil.hashPassword(psw, user.getSalt()))) {
+                // user.setRoles(getRoles(user.getId()));
```
Comment (Sofasmile) [inline src/main/java/internet/shop/dao/hibernate/UserDaoHibernateImpl.java]: remove commented code from repository

File: README.md (hunk 1 above)
Comment (tarasulo) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (tarasulo) [reply README.md:2]: correctly

File: settings.xml (hunk 3 above)
Comment (tarasulo) [reply settings.xml]: done

File: src/main/java/internet/shop/controller/AddToBucketController.java (hunk 4 above)
Comment (tarasulo) [reply src/main/java/internet/shop/controller/AddToBucketController.java]: done

File: src/main/java/internet/shop/dao/hibernate/BucketDaoHibernateImpl.java (hunk 5 above)
Comment (tarasulo) [reply src/main/java/internet/shop/dao/hibernate/BucketDaoHibernateImpl.java]: fixed

File: src/main/java/internet/shop/dao/hibernate/UserDaoHibernateImpl.java (hunk 6 above)
Comment (tarasulo) [reply src/main/java/internet/shop/dao/hibernate/UserDaoHibernateImpl.java:47]: done

File: src/main/java/internet/shop/dao/hibernate/UserDaoHibernateImpl.java (hunk 7 above)
Comment (tarasulo) [reply src/main/java/internet/shop/dao/hibernate/UserDaoHibernateImpl.java]: done

**Review by Sofasmile — APPROVED** [review]

File: src/main/java/internet/shop/controller/AddToBucketController.java [hunk 8]
```diff
@@ -2,27 +2,40 @@
...
 import javax.servlet.http.HttpServletResponse;
+
 import org.apache.log4j.Logger;
 
 public class AddToBucketController extends HttpServlet {
+
     private static Logger logger = Logger.getLogger(AddToBucketController.class);
     @Inject
-    static BucketService bucketService;
+    private static BucketService bucketService;
```
Comment (kolyansv) [inline src/main/java/internet/shop/controller/AddToBucketController.java:22]: why do we need 'static' keyword in all places where you inject service?

File: src/main/java/internet/shop/dao/jdbc/UserDaoJdbcImpl.java [hunk 9]
```diff
@@ -34,29 +40,44 @@ public UserDaoJdbcImpl(Connection connection) {
...
+                addRole(user.getId(), role);
+            }
         } catch (SQLException e) {
             logger.error("Can't create user", e);
         }
         return user;
     }
 
+    private Role addRole(Long userId, Role role) {
+        String query = "INSERT INTO store.user_role (user_id, role_id) VALUES (?, ?);";
```
Comment (kolyansv) [inline src/main/java/internet/shop/dao/jdbc/UserDaoJdbcImpl.java]: extract all queries to object variables

File: src/main/java/internet/shop/dao/jdbc/UserDaoJdbcImpl.java [hunk 10]
```diff
@@ -34,29 +40,44 @@ public UserDaoJdbcImpl(Connection connection) {
 
     @Override
     public User add(User user) {
-        String query = "INSERT INTO users (name, login, password, token, salt) "
+        String query = "INSERT INTO store.users (name, login, password, token, salt) "
```
Comment (kolyansv) [inline src/main/java/internet/shop/dao/jdbc/UserDaoJdbcImpl.java]: What is the reason to use database name here?

File: src/main/java/internet/shop/dao/hibernate/BucketDaoHibernateImpl.java [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+    private static Logger logger = Logger.getLogger(BucketDaoHibernateImpl.class);
+    @Inject
+    private static ItemDao itemDao;
+
+    @Override
+    public Bucket create(Bucket bucket) {
+        Transaction transaction = null;
+        Long bucketId = null;
+        Session session = null;
+        try {
```
Comment (kolyansv) [inline src/main/java/internet/shop/dao/hibernate/BucketDaoHibernateImpl.java:29]: in some cases you're using try-with-resources to release resources. In the other cases you're using plain try with finally block. Update you code to follow the same approach (doesn't matter which)

File: src/main/java/internet/shop/lib/AnnotatedClassMap.java [hunk 12]
```diff
@@ -10,11 +10,12 @@
 import internet.shop.service.ItemService;
 import internet.shop.service.OrderService;
 import internet.shop.service.UserService;
-
 import java.util.HashMap;
 import java.util.Map;
+import org.apache.log4j.Logger;
 
 public class AnnotatedClassMap {
+    private static final Logger logger = Logger.getLogger(AnnotatedClassMap.class);
```
Comment (kolyansv) [inline src/main/java/internet/shop/lib/AnnotatedClassMap.java]: Due to java convention all constants should be written with upper-case letters

Comment (tarasulo) [reply src/main/java/internet/shop/lib/AnnotatedClassMap.java]: done

File: src/main/java/internet/shop/dao/hibernate/BucketDaoHibernateImpl.java (hunk 11 above)
Comment (tarasulo) [reply src/main/java/internet/shop/dao/hibernate/BucketDaoHibernateImpl.java:29]: done

File: src/main/java/internet/shop/dao/jdbc/UserDaoJdbcImpl.java (hunk 10 above)
Comment (tarasulo) [reply src/main/java/internet/shop/dao/jdbc/UserDaoJdbcImpl.java]: without a database name, it doesn't work ((

File: src/main/java/internet/shop/dao/jdbc/UserDaoJdbcImpl.java (hunk 9 above)
Comment (tarasulo) [reply src/main/java/internet/shop/dao/jdbc/UserDaoJdbcImpl.java]: done

File: src/main/java/internet/shop/controller/AddToBucketController.java (hunk 8 above)
Comment (tarasulo) [reply src/main/java/internet/shop/controller/AddToBucketController.java:22]: without the static keyword, it doesn't work. Static fields will be injected after the first object of the class was created, which means no access to the static field in the constructor

**Pull request merged**